py/server/deephaven/table_listener.py

```diff
@@ -555,6 +516,7 @@ def stop(self) -> None:
         if not self.started:
             return
 
+        # self.tables[0] is guaranteed to be a refreshing table
```

Comment:
This is not the comment you really need here. You (additionally) need a comment explaining that we should lock, in order to ensure that all the listeners are removed on the same cycle.